### ISSUE TITLE
requirejs confusion

### ISSUE THREAD

**Issue by GordonSmith** [cpettitt/graphlib#32]
Am I correct in saying that graphlib 1.0.1 does not support loading via requirejs, but dagre does?

**Comment by sprilukin** [comment]
@cpettitt first of all i want to thank you for such a great library!

I think it's important to add info about how to made this library umd compatible to a wiki.
And also it would be great to add umd build to a dist folder, for example as dist/graphlib.umd.js (ideally with update of bower.json)

Not all people know how to or want run browserify manually to get self contained umd/amd compatible build. Even people who know how to do this would prefer use your umd build rather than make fork just to push their own umd build.

Again - thanks a lot for the library!

**Comment by GordonSmith** [comment]
(as a complete aside - google is using this library + dagre with TensorFlow:  https://www.tensorflow.org/)

**Comment by cpettitt** [comment]
@sprilukin IIRC, the -s graphlib approach would work for all use cases, so could just be the default - I think that's what I ended up doing for dagre. This is just waiting from a PR from an interested enough party who can verify with a UMD loader.

@GordonSmith yeah, I saw that. Pretty cool! Thanks for mentioning it.

**Comment by sprilukin** [comment]
Well i've already made my build with -s graphlib and can confirm that it works perfectly in browser with amd.  So we just need to test this in node and in browser without amd. I'll try to do that.
Or how you expect to do such verification?

**Comment by cpettitt** [comment]
That part is easy: just run the tests (`make test` should do it). If you have a patch, I can run the tests and publish a new version.

**Comment by sprilukin** [comment]
Well the problem is that i'm not familiar with make,
i just build umd version for myself using direct command:
```sh
browserify index.js -o dist/graphlib.umd.js -s graphlib
```

i just tried to modify browserify targets in makefile so they'll produce same result,
but result is different and tests are failed.

here are my changes in makefile:

```diff
diff --git a/Makefile b/Makefile
index f05dd9f..759525b 100644
--- a/Makefile
+++ b/Makefile
@@ -52,14 +52,14 @@ browser-test: $(BUILD_DIR)/$(MOD).js $(BUILD_DIR)/$(MOD).core.js
 bower.json: package.json src/release/make-bower.json.js
 	@src/release/make-bower.json.js > $@
 
-$(BUILD_DIR)/$(MOD).js: browser.js $(SRC_FILES) | unit-test
-	@$(BROWSERIFY) $< > $@
+$(BUILD_DIR)/$(MOD).js: index.js $(SRC_FILES) | unit-test
+	@$(BROWSERIFY) $< > $@ -s graphlib
 
 $(BUILD_DIR)/$(MOD).min.js: $(BUILD_DIR)/$(MOD).js
 	@$(UGLIFY) $< --comments '@license' > $@
 
 $(BUILD_DIR)/$(MOD).core.js: browser.js $(SRC_FILES) | unit-test
-	@$(BROWSERIFY) $< > $@ --no-bundle-external
+	@$(BROWSERIFY) $< > $@ --no-bundle-external -s graphlib
 
 $(BUILD_DIR)/$(MOD).core.min.js: $(BUILD_DIR)/$(MOD).core.js
 	@$(UGLIFY) $< --comments '@license' > $@
```

**Comment by sprilukin** [comment]
Ok, PR #57 should do the trick, both graphlib and graphlib.core are now amd compatible with this PR,
see [this](https://github.com/cpettitt/graphlib/pull/57#issuecomment-186636883) comment for details about using graphlib.core.

**Comment by cpettitt** [comment]
Fixed in v2.1.0. Thanks @sprilukin!

**Comment by sprilukin** [comment]
Glad i can help!
